### PR TITLE
exposed varargs methods in JPQLQueryFactory

### PR DESCRIPTION
**Pull request by lpandzic**
Fixes issue #1474

### CODE REVIEW

**Comment by lpandzic** [comment]
On a second thought, I'm not sure this makes sense in selectFrom since it modifies projection and selection together. What do you think?

**Comment by timowest** [comment]
I agree that they are not that useful for `selectFrom`.

File: querydsl-jpa/src/main/java/com/querydsl/jpa/hibernate/HibernateQueryFactory.java [hunk 1]
```diff
@@ -101,11 +102,21 @@ public HibernateDeleteClause delete(EntityPath<?> path) {
     }
 
     @Override
+    public <T> JPQLQuery<Tuple> selectFrom(EntityPath<T>... from) {
```
Comment (timowest) [inline querydsl-jpa/src/main/java/com/querydsl/jpa/hibernate/HibernateQueryFactory.java]: return type `HibernateQuery<Tuple>`

File: querydsl-jpa/src/main/java/com/querydsl/jpa/impl/JPAQueryFactory.java [hunk 2]
```diff
@@ -112,11 +113,21 @@ public JPADeleteClause delete(EntityPath<?> path) {
     }
 
     @Override
+    public <T> JPQLQuery<Tuple> selectFrom(EntityPath<T>... from) {
```
Comment (timowest) [inline querydsl-jpa/src/main/java/com/querydsl/jpa/impl/JPAQueryFactory.java]: return type `JPAQuery<Tuple>`

File: querydsl-jpa/src/main/java/com/querydsl/jpa/impl/JPAQueryFactory.java [hunk 3]
```diff
@@ -112,11 +113,21 @@ public JPADeleteClause delete(EntityPath<?> path) {
...
+        return select(from).from(from);
+    }
+
+    @Override
     public JPAQuery<?> from(EntityPath<?> from) {
         return query().from(from);
     }
 
     @Override
+    public JPQLQuery<?> from(EntityPath<?>... from) {
```
Comment (timowest) [inline querydsl-jpa/src/main/java/com/querydsl/jpa/impl/JPAQueryFactory.java]: return type `JPAQuery<?>`

**Comment by lpandzic** [comment]
I've removed selectFrom method and specialized return types.

**Comment by lpandzic** [comment]
Removed unused imports. :)

**Comment by timowest** [comment]
This looks good to me.

**Comment by Shredder121** [comment]
This looks clean to me.
@johnktims you can merge if you agree.

@lpandzic thanks for the pull request!

**Comment by johnktims** [comment]
Looks good to me. Thanks for the PR, @lpandzic!

**Pull request merged**